### PR TITLE
Add Toast notification to ExternalLinkButton's context menu

### PR DESCRIPTION
**Pull request by andrewhong04**
Continuation from: https://github.com/ppy/osu/pull/19375
image

### CODE REVIEW

File: osu.Game/Graphics/UserInterface/ExternalLinkButton.cs [hunk 1]
```diff
@@ -44,6 +50,23 @@ public ExternalLinkButton(string link = null)
             };
         }
 
+        private class CopyUrlToast : Toast
+        {
+            public CopyUrlToast(LocalisableString value)
+                : base(UserInterfaceStrings.GeneralHeader, value, "")
+            {
+            }
+        }
```
Comment (frenzibyte) [inline osu.Game/Graphics/UserInterface/ExternalLinkButton.cs:59]: Nested classes go to bottom.

File: osu.Game/Graphics/UserInterface/ExternalLinkButton.cs [hunk 2]
```diff
@@ -44,6 +50,23 @@ public ExternalLinkButton(string link = null)
...
+
+        private void copyUrl()
+        {
+            if (Link != null)
+            {
+                host.GetClipboard()?.SetText(Link);
+                onScreenDisplay?.Display(new CopyUrlToast(ToastStrings.CopiedUrl));
+            }
```
Comment (frenzibyte) [inline osu.Game/Graphics/UserInterface/ExternalLinkButton.cs:67]: Link is guaranteed not to be null from `ContextMenuItems`.
```suggestion
            host.GetClipboard()?.SetText(Link);
            onScreenDisplay?.Display(new CopyUrlToast(ToastStrings.CopiedUrl));
```

File: osu.Game/Graphics/UserInterface/ExternalLinkButton.cs [hunk 3]
```diff
@@ -53,7 +76,7 @@ public MenuItem[] ContextMenuItems
                 if (Link != null)
                 {
                     items.Add(new OsuMenuItem("Open", MenuItemType.Standard, () => host.OpenUrlExternally(Link)));
-                    items.Add(new OsuMenuItem("Copy URL", MenuItemType.Standard, () => host.GetClipboard()?.SetText(Link)));
+                    items.Add(new OsuMenuItem("Copy URL", MenuItemType.Standard, () => copyUrl()));
```
Comment (frenzibyte) [inline osu.Game/Graphics/UserInterface/ExternalLinkButton.cs:79]:
```suggestion
                    items.Add(new OsuMenuItem("Copy URL", MenuItemType.Standard, copyUrl));
```

File: osu.Game/Localisation/ToastStrings.cs [hunk 4]
```diff
@@ -44,6 +44,11 @@ public static class ToastStrings
         /// </summary>
         public static LocalisableString SkinSaved => new TranslatableString(getKey(@"skin_saved"), @"Skin saved");
 
+        /// <summary>
+        /// "Copied URL"
+        /// </summary>
+        public static LocalisableString CopiedUrl => new TranslatableString(getKey(@"copied_url"), @"Copied URL");
```
Comment (frenzibyte) [inline osu.Game/Localisation/ToastStrings.cs:50]: Probably want this to be "URL copied" inline with the two messages above.

**Review by frenzibyte — CHANGES_REQUESTED** [review]

**Review by peppy — APPROVED** [review]

**Pull request merged**